lib/api/route/typing.dart

```diff
@@ -0,0 +1,26 @@
+import '../core.dart';
+import '../model/events.dart';
+import 'messages.dart';
+
+
+/// https://zulip.com/api/set-typing-status
+Future<void> setTypingStatus(ApiConnection connection, {
+  required TypingOp op,
+  required MessageDestination destination,
+}) {
+  final supportsTypeChannel = connection.zulipFeatureLevel! >= 248; // TODO(server-9)
+  switch (destination) {
+    case StreamDestination():
+      return connection.post('setTypingStatus', (_) {}, 'typing', {
+        'type': RawParameter((supportsTypeChannel) ? 'channel' : 'stream'),
+        'op': RawParameter(op.toJson()),
+        'stream_id': destination.streamId,
+        'topic': RawParameter(destination.topic),
+      });
+    case DmDestination():
+      return connection.post('setTypingStatus', (_) {}, 'typing', {
+        'op': RawParameter(op.toJson()),
+        'to': destination.userIds,
+      });
```

Comment:
What's the effect of not passing 'type' in the DM case? From the doc, it looks like the preferred client behavior (assuming Zulip Server 4+) is to pass 'private' when FL is <174, else 'direct'. I'm not sure I've read closely enough to know what happens across the range of supported servers if we just don't pass 'type' at all.

Reply:
This was intended to match the mobile apps' implementation, where we omit the type field for dm typing notfications I think it is not clear that the default was "private" back then. Skipping `type` this way allows us to not worry about the "private" -> "direct" transition of this API.

See: https://chat.zulip.org/#narrow/stream/378-api-design/topic/stream.20typing.20indicator

Reply:
I see. I guess since zulip-mobile does that not-preferred thing and presumably still works, probably makes it harmless if we also do the not-preferred thing. 🙂 But also, the burden of handling the "private" -> "direct" transition isn't heavy—

> pass 'private' when FL is <174, else 'direct'

—so I'd be in favor of just doing that. If not, though, let's include a comment to make it clear why we're doing something that's different from the thing that seems natural/expected from a reading of the API doc.

Reply:
Yeah, cleanest to just do the thing that's preferred in the current API.

(And then adapt for old servers as necessary — but "type" has been accepted since server-4, so effectively forever now.)